rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -266,252 +246,112 @@ private MultipartEntity createLoginPasswordSSHKeyMultipart(String login, String
      * (non-Javadoc)
      * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#createNodeSource(java.lang.String, java.lang.String, java.lang.String, java.lang.String[], java.lang.String[], java.lang.String, java.lang.String[], java.lang.String[])
      */
-    public String createNodeSource(String sessionId, String nodeSourceName, String infrastructureType,
-            String[] infrastructureParameters, String[] infrastructureFileParameters, String policyType,
-            String[] policyParameters, String[] policyFileParameters) throws RestServerException,
+    public String createNodeSource(final String sessionId, final String nodeSourceName, final String infrastructureType,
```

Comment:
The compiler will grumble if there are not final because they are used in an anonymous class in the following.